gobblin-metrics/src/main/java/gobblin/metrics/kafka/KafkaReporter.java

```diff
@@ -357,72 +390,48 @@ public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> c
    * @param value
    * @param path suffixes to more precisely identify the meaning of the reported value
-   * @return Singleton list of byte arrays representing the value
+   * @return a Singleton list of {@link gobblin.metrics.Metric}.
    */
-  protected List<byte[]> serializeSingleValue(String name, Object value, String... path) {
-    List<byte[]> metrics = new ArrayList<byte[]>();
+  protected List<Metric> serializeSingleValue(String name, Number value, String... path) {
+    List<Metric> metrics = new ArrayList<Metric>();
     metrics.add(serializeValue(name, value, path));
     return metrics;
   }
 
   /**
-   * Serializes a single metric key-value pair to send to Kafka.
+   * Converts a single key-value pair into a metric.
    * @param name name of the metric
    * @param value value of the metric to report
    * @param path additional suffixes to further identify the meaning of the reported value
-   * @return a byte array containing the key-value pair representing the metric
+   * @return a {@link gobblin.metrics.Metric}.
    */
-  protected byte[] serializeValue(String name, Object value, String... path) {
-    String str = stringifyValue(name, value, path);
-    if(!Strings.isNullOrEmpty(str)) {
-      return str.getBytes(Charsets.UTF_8);
-    } else {
-      return null;
-    }
+  protected Metric serializeValue(String name, Number value, String... path) {
+    return new Metric(MetricRegistry.name(name, path), value.doubleValue());
   }
 
   /**
-   * Converts a single metric key-value pair to a string.
-   * @param name name of the metric
-   * @param value value of the metric to report
-   * @param path additional suffixes to further identify the meaning of the reported value
-   * @return a string containing the key-value pair representing the metric
+   * Converts a {@link gobblin.metrics.MetricReport} to bytes to send through Kafka.
+   * @param report MetricReport to serialize.
+   * @return Serialized bytes.
    */
-  protected synchronized  String stringifyValue(String name, Object value, String... path) {
-    Metric metric = new Metric();
-    metric.name = MetricRegistry.name(name, path);
-    metric.value = value;
-    metric.tags = tags;
-    metric.timestamp = System.currentTimeMillis();
-
-    try {
-      return mapper.writeValueAsString(metric);
-    } catch(Exception e) {
-      // If there is actually something wrong with the serializer,
-      // this exception would be thrown for every single metric serialized.
-      // Instead, report at warn level at most every 10 seconds.
-      LOGGER.trace("Could not serialize Avro record for Kafka Metrics. Exception: %s", e.getMessage());
-      if(System.currentTimeMillis() - lastSerializeExceptionTime > 10000) {
-        LOGGER.warn("Could not serialize Avro record for Kafka Metrics. Exception: %s", e.getMessage());
-        lastSerializeExceptionTime = System.currentTimeMillis();
-      }
+  protected synchronized byte[] serializeReport(MetricReport report) {
```

Comment:
Please add an id (int) before the schema to have a path for schema migration later.